File: 第1讲--RAC初探/预习资料/06--代码基础/001---RAC底层分析/001---RAC底层分析/ViewController.m
//
//  ViewController.m
//  001---RAC底层分析
//
//  Created by Cooci on 2018/8/2.
//  Copyright © 2018年 Cooci. All rights reserved.
//

#import "ViewController.h"
#import <ReactiveObjC.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    // RAC 流程分析
    // 1: 信号产生
    RACSignal *signal = [RACSignal createSignal:^RACDisposable * _Nullable(id<RACSubscriber>  _Nonnull subscriber) {
        
        // 3:发送信号
        [subscriber sendNext:@"Cooci"];
        
        return [RACDisposable disposableWithBlock:^{
            NSLog(@"我们销毁了");
        }];
    }];
    // 2: 订阅信号
    [signal subscribeNext:^(id  _Nullable x) {
        NSLog(@"订阅到了:%@",x);
    }];
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 第3讲--RAC底层原理分析下/000---RxSwift总结应用/001--RxSwift深入浅出/ViewController.swift
//
//  ViewController.swift
//  001--RxSwift深入浅出
//
//  Created by Cooci on 2018/5/26.
//  Copyright © 2018年 Cooci. All rights reserved.
//

import UIKit
import RxSwift
import RxCocoa

class ViewController: UIViewController {

    let disposB = DisposeBag()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        testUI()
        testRxSwiftCreat()
    }
    
    func testRxSwiftCreat(){

    }
    
    func testUI(){
        let textf = UITextField(frame: CGRect(x: 50, y: 50, width: 100, height: 30))
        textf.borderStyle = UITextBorderStyle.roundedRect
        self.view.addSubview(textf)
        
        let label = UILabel(frame: CGRect(x: 50, y: 90, width: 200, height: 30))
        label.textColor = UIColor.orange
        self.view.addSubview(label)
        
        let btn = UIButton(type: .system)
        btn.frame = CGRect(x: 50, y: 120, width: 100, height: 30)
        btn.setTitle("登录", for:UIControlState.normal)
        self.view.addSubview(btn)
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

enum MyError:Error {
    case errorA
    case errorB
    var errorType: String{
        switch self {
        case .errorA:
            return "this is A error"
        case .errorB:
            return "this is B error"
        }
    }
    
}



